# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_conic_gradient.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_conic_gradient.glsl
--- a/gfx/wr/webrender/res/cs_conic_gradient.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_conic_gradient.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -6,13 +6,13 @@
 
 #define PI                  3.141592653589793
 
-varying vec2 v_pos;
+varying highp vec2 v_pos;
 
-flat varying vec2 v_center;
+flat varying highp vec2 v_center;
 
 // x: start offset, y: offset scale, z: angle
 // Packed in to a vector to work around bug 1630356.
-flat varying vec3 v_start_offset_offset_scale_angle_vec;
+flat varying highp vec3 v_start_offset_offset_scale_angle_vec;
 #define v_start_offset v_start_offset_offset_scale_angle_vec.x
 #define v_offset_scale v_start_offset_offset_scale_angle_vec.y
 #define v_angle v_start_offset_offset_scale_angle_vec.z